protocols/bftcosi/bftcosi.go

```diff
@@ -3,33 +3,38 @@
 package bftcosi
 
 import (
-	"math"
+	"crypto/sha512"
 	"sync"
 
-	"github.com/dedis/cothority/cosi"
+	"gopkg.in/dedis/cothority.v0/lib/dbg"
+
 	"github.com/dedis/cothority/log"
 	"github.com/dedis/cothority/sda"
 	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/cosi"
 )
 
 // VerificationFunction can be passes to each protocol node. It will be called
 // (in a go routine) during the (start/handle) challenge prepare phase of the
 // protocol. The passed message is the same as sent in the challenge phase.
-type VerificationFunction func(Msg []byte) bool
+type VerificationFunction func(Msg []byte, Data []byte) bool
 
 // ProtocolBFTCoSi is the main struct for running the protocol
 type ProtocolBFTCoSi struct {
 	// the node we are represented-in
 	*sda.TreeNodeInstance
+	// The message that will be signed by the BFTCosi
 	Msg []byte
+	// Data going along the msg to the verification
+	Data []byte
```

Comment:
Question: I do not really understand why this field is necessary. (Why would it be useful for example for ByzCoin?)